File: app/src/main/java/com/example/weatherapplication/WatchList/WatchListWeather.java
package com.example.weatherapplication.WatchList;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.Index;
import androidx.room.PrimaryKey;

import java.util.Calendar;

@Entity(tableName = "table_watch_lists")
public class WatchListWeather {
    @PrimaryKey
    @NonNull
    private String cityName;
    private String country;
    private String description;
    private String temperature;
    private String weatherImagePath;
    private double lat;
    private double lon;
    private long createdTime;

    public WatchListWeather(String cityName, String country, String description, String temperature
                                        ,String weatherImagePath, double lat,double lon,long createdTime) {
        this.description = description;
        this.temperature = temperature;
        this.cityName = cityName;
        this.weatherImagePath = weatherImagePath;
        this.lat = lat;
        this.lon = lon;
        this.country = country;
        this.createdTime = createdTime;
    }

    public String getDescription() {
        return description;
    }

    public String getTemperature() {
        return temperature;
    }

    public String getCityName() {
        return cityName;
    }

    public String getWeatherImagePath() {
        return weatherImagePath;
    }

    public double getLat() {
        return lat;
    }

    public double getLon() {
        return lon;
    }

    public void setCreatedTime(long createdTime) {
        this.createdTime = createdTime;
    }

    public long getCreatedTime() {
        return createdTime;
    }

    public void setCityName(String cityName) {
        this.cityName = cityName;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public void setTemperature(String temperature) {
        this.temperature = temperature;
    }

    public void setWeatherImagePath(String weatherImagePath) {
        this.weatherImagePath = weatherImagePath;
    }

    public void setLat(double lat) {
        this.lat = lat;
    }

    public void setLon(double lon) {
        this.lon = lon;
    }

    public String getCountry() {
        return country;
    }
}

File: app/src/main/java/com/amirreza/domain/entity/WatchListWeather.kt
package com.amirreza.domain.entity

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.amirreza.domain.entity.OneCallWeatherEntitys.CityWeatherAllInformation
import java.util.*

@Entity(tableName = "table_watch_lists")
class WatchListWeather(
    @field:PrimaryKey var cityName: String,
    var country: String,
    var description: String,
    var temperature: String,
    var weatherImagePath: String,
    var lat: Double,
    var lon: Double,
    var createdTime:Long= Calendar.getInstance().time.time
){
    companion object{
        fun convertCityWeatherAllDataToWatchListWeather(city:String,country:String,all:CityWeatherAllInformation):WatchListWeather{
            return WatchListWeather(
                cityName = city,
                country = country,
                description =  all.current.timeDescription,
                temperature = all.current.temperature.toString(),
                weatherImagePath = all.getWeatherImagePath().toString(),
                lat = all.lat,
                lon = all.lon,
            )
        }
    }
}